pkg/adt/interval_tree.go

```diff
@@ -486,14 +546,41 @@ func (ivt *IntervalTree) Stab(iv Interval) (ivs []*IntervalValue) {
 }
 
 // Union merges a given interval tree into the receiver.
-func (ivt *IntervalTree) Union(inIvt IntervalTree, ivl Interval) {
+func (ivt *IntervalTree) Union(inIvt *IntervalTree, ivl Interval) {
 	f := func(n *IntervalValue) bool {
 		ivt.Insert(n.Ivl, n.Val)
 		return true
 	}
 	inIvt.Visit(ivl, f)
 }
 
+func (ivt *IntervalTree) LevelOrder() [][]*intervalNode {
```

Comment:
is this only used by test? if so, move this to test file.

Reply:
Yes, I will change.